DQMOffline/L1Trigger/src/L1TStage2CaloLayer2Offline.cc

```diff
@@ -0,0 +1,596 @@
+    edm::EventSetup const& context)
+{
+  edm::LogInfo("L1TStage2CaloLayer2Offline") << "L1TStage2CaloLayer2Offline::beginLuminosityBlock" << std::endl;
+}
+
+//
+// -------------------------------------- Analyze --------------------------------------------
+//
+void L1TStage2CaloLayer2Offline::analyze(edm::Event const& e, edm::EventSetup const& eSetup)
+{
+  edm::LogInfo("L1TStage2CaloLayer2Offline") << "L1TStage2CaloLayer2Offline::analyze" << std::endl;
+
+  edm::Handle<reco::VertexCollection> vertexHandle;
+  e.getByToken(thePVCollection_, vertexHandle);
+  if (!vertexHandle.isValid()) {
+    edm::LogError("L1TStage2CaloLayer2Offline") << "invalid collection: vertex " << std::endl;
+    return;
+  }
+
+  unsigned int nVertex = vertexHandle->size();
+  dqmoffline::l1t::fillWithinLimits(h_nVertex_, nVertex);
+
+  // L1T
+  fillEnergySums(e, nVertex);
+  fillJets(e, nVertex);
+}
+
+void L1TStage2CaloLayer2Offline::fillEnergySums(edm::Event const& e, const unsigned int nVertex)
+{
+  edm::Handle<l1t::EtSumBxCollection> l1EtSums;
+  e.getByToken(stage2CaloLayer2EtSumToken_, l1EtSums);
+
+  edm::Handle<reco::CaloJetCollection> caloJets;
+  e.getByToken(theCaloJetCollection_, caloJets);
+
+  edm::Handle<reco::CaloMETCollection> caloMETs;
+  e.getByToken(thecaloMETCollection_, caloMETs);
+
+  if (!caloJets.isValid()) {
+    edm::LogError("L1TStage2CaloLayer2Offline") << "invalid collection: calo jets " << std::endl;
+    return;
+  }
+  if (!caloMETs.isValid()) {
+    edm::LogError("L1TStage2CaloLayer2Offline") << "invalid collection: Offline E_{T}^{miss} " << std::endl;
+    return;
+  }
+  if (!l1EtSums.isValid()) {
+    edm::LogError("L1TStage2CaloLayer2Offline") << "invalid collection: L1 ET sums " << std::endl;
+    return;
+  }
+
+  int bunchCrossing = 0;
+
+  double l1MET(0);
+  double l1METPhi(0);
+  double l1MHT(0);
+  double l1MHTPhi(0);
+  double l1ETT(0);
+  double l1HTT(0);
+
+  for (auto etSum = l1EtSums->begin(bunchCrossing); etSum != l1EtSums->end(bunchCrossing); ++etSum) {
+    double et = etSum->et();
+    double phi = etSum->phi();
+
+    switch (etSum->getType()) {
+    case l1t::EtSum::EtSumType::kMissingEt:
+      l1MET = et;
+      l1METPhi = phi;
+      break;
+    case l1t::EtSum::EtSumType::kTotalEt:
+      l1ETT = et;
+      break;
+    case l1t::EtSum::EtSumType::kMissingHt:
+      l1MHT = et;
+      l1MHTPhi = phi;
+      break;
+    case l1t::EtSum::EtSumType::kTotalHt:
+      l1HTT = et;
+    default:
+      break;
+    }
+
+  }
+
+  double recoMET(caloMETs->front().et());
+  double recoMETPhi(caloMETs->front().phi());
+  double recoMHT(0);
+  double recoMHTPhi(0);
+  double recoETT(0);
+  double recoHTT(0);
+
+  TVector2 mht(0., 0.);
+
+  for (auto jet = caloJets->begin(); jet != caloJets->end(); ++jet) {
+    double et = jet->et();
+    if (et < 30) {
+      continue;
+    }
+    TVector2 jetVec(0., 0.);
+    jetVec.SetMagPhi(et, jet->phi());
```

Comment:
I do not understand.
Do you mean `jetVec.Set(et*jet->x(), et*jet->y())` (github garbles the `*`)?
I get the error `const class reco::CaloJet' has no member named 'x'`

Or do you mean
`jetVec.Set(et*cos(jet->phi()), et*sin(jet->phi()))`?